### DISCUSSION TITLE
Socket io v2 to v3: Default namespace is not connected anymore

### DISCUSSION THREAD

**Discussion by SamTV12345** [socketio/socket.io#4953] [Q&A] [answered]
We at Etherpad have the problem that the default namespace isn't connected if I connect to another namespace. Thus the normal sessionMiddlewares aren't being called. Do you know a way how to get this behaviour back?

**Comment by darrachequesne** [comment, marked as answer]
Hi!

This is indeed a pretty big change when migrating from v2 to v3. 

You should be able to register the middleware on your other namespace:

```js
io.of("/my-other-namespace").use(sessionMiddleware);
```

Reference: https://socket.io/docs/v4/server-api/#namespaceusefn

See also: https://socket.io/how-to/register-a-global-middleware

**Reply by SamTV12345** [reply]
Thanks for the help :). It is indeed a big change. That solved the problem. Now we just got a problem with our websockets. Do you have an idea why? This was the pr that upgraded to v3: https://github.com/ether/etherpad-lite/commit/b2be2ca71464f4da4852b9218d871adee04d194d . Once Etherpad starts, it uses long polling, tries to switch over to websockets, connects and then returns bad request 400.

**Reply by darrachequesne** [reply]
HTTP 400 usually means something went wrong and the connection was discarded by the server. Were you able to debug the issue? You can display the logs with the `DEBUG` env variable: `DEBUG=socket.io*,engine.io*`

**Reply by SamTV12345** [reply]
I get the following messages:

socket.io:client close with reason transport error
socket.io:socket closing socket - reason transport error

The server and the client have transports websocket and polling enabled.

**Reply by darrachequesne** [reply]
@SamTV12345 were you able to find the culprit?

**Reply by SamTV12345** [reply]
Yes. I was. I now only have the problem that the socket randomly disconnects. I'll debug that further and report back if I solve that problem.
